core/src/main/java/org/elasticsearch/search/internal/ShardSearchRequest.java

```diff
@@ -60,6 +60,18 @@
     Scroll scroll();
 
     /**
+     * Sets if this shard search needs to be profiled or not
+     * @param profile True if the shard should be profiled
+     */
+    void profile(boolean profile);
+
+    /**
+     * Returns if this shard search is being profiled or not
```

Comment:
`Returns <code>true</code> iff this shard search is profiled, otherwise <code>false</code>